gibbon-rts/rts-c/gibbon_rts.c

```diff
@@ -1923,13 +1929,21 @@ int gib_init(int argc, char **argv)
             gib_show_usage(argv);
             exit(0);
         }
-        else if (strcmp(argv[i], "--biginf-buffer-size") == 0 && i < argc - 1)
-        {
+        else if (strcmp(argv[i], "--biginf-buffer-size") == 0 && i < argc - 1){   
+            if (i+1 >= argc) {
+                fprintf(stderr, "Not enough arguments after --biginf-buffer-size, expected <int>.\n");
+                gib_show_usage(argv);
+                exit(1);
```

Comment:
You keep repeating this `if` statement for every exceptional case below. Do you want to factor it out into a utility function? This will save you a bunch of repetition. Also, all of them serve the case of `"Not enough arguments"`, so you will only need to pass the name of the parameter, `argc`,` and `argv` to the utility function.